lib/spack/spack/package.py

```diff
@@ -380,6 +380,22 @@ def __init__(self, spec):
         if not hasattr(self, 'list_depth'):
             self.list_depth = 1
 
+        # Set default licensing information
+        if not hasattr(self, 'license_required'):
+            self.license_required = False
+
+        if not hasattr(self, 'license_comment'):
+            self.license_comment = '#'
+
+        if not hasattr(self, 'license_files'):
+            self.license_files = []
+
+        if not hasattr(self, 'license_vars'):
+            self.license_vars = []
+
+        if not hasattr(self, 'license_url'):
+            self.license_url = None
```

Comment:
Spack could probably do a better job with instance vs. class attributes in `Package`.  Basically `spack.repo.get('<pkgname>')` is what Spack typically uses to get an instance just for package-level metadata.  The `Package` constructor above just does a few sanity checks to make sure subclasses set the right variables.  We could probably clean this up a bit.